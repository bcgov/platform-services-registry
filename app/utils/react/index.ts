export * from './hooks';
export * from './search-params';
