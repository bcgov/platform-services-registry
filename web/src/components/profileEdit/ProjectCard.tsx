//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import React from 'react';
import { Box, Flex, Text } from 'rebass';
import theme from '../../theme';

interface IProjectCardProps {
  projectDetails: ProjectDetails;
}

export interface ProjectDetails {
  id?: string;
  name?: string;
  description?: string;
  prioritySystem?: string;
  busOrgId?: string;
  ministryName?: string;
  primaryClusterDisplayName?: string;
  other?: string;
  migratingLicenseplate?: string;
}

const ProjectCard: React.FC<IProjectCardProps> = (props) => {
  const {
    projectDetails: {
      name = '',
      description = '',
      ministryName = '',
      migratingLicenseplate = '',
      primaryClusterDisplayName = '',
    },
  } = props;
  return (
    <Flex flexWrap="wrap">
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="h3">Project Name</Text>
        <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
          This is the public name of the application, please avoid acronyms;
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
          {name}
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="h3">Project Description</Text>
        <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
          This is a brief description of your project;
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
          {description}
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="h3">Ministry</Text>
        <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
          This is the ministry that is responsible for your application;
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
          {ministryName}
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="h3">Cluster</Text>
        <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
          This is the cluster this namespace is provisioned in;
        </Text>
      </Box>
      <Box width={1 / 2} px={2} mt={3}>
        <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
          {primaryClusterDisplayName}
        </Text>
      </Box>
      {migratingLicenseplate ? (
        <Flex width={1}>
          <Box width={1 / 2} px={2} mt={3}>
            <Text as="h3">Migrating Application</Text>
            <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
              This is the license plate of your migrating OCP 3.11 application;
            </Text>
          </Box>
          <Box width={1 / 2} px={2} mt={3}>
            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
              {migratingLicenseplate}
            </Text>
          </Box>
        </Flex>
      ) : (
        <Flex />
      )}
    </Flex>
  );
};

export default ProjectCard;
