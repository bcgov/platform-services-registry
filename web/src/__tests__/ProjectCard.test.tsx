//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import { render } from '@testing-library/react';
import React from 'react';
import ProjectCard from '../components/profileEdit/ProjectCard';

test('<ProjectCard /> should render without Migration Licenseplate', () => {
  const stubPropTitle = 'Health Gateway';
  const stubPropTextBody = 'This is a test description for health gateway app';
  const stubPropMinistry = "Citizen's Services";

  const { container } = render(
    <ProjectCard
      projectDetails={{
        name: stubPropTitle,
        description: stubPropTextBody,
        ministryName: stubPropMinistry,
      }}
    />,
  );

  expect(container).toMatchSnapshot();
});

test('<ProjectCard /> should render with Migration Licenseplate', () => {
  const stubPropTitle = 'Health Gateway';
  const stubPropTextBody = 'This is a test description for health gateway app';
  const stubPropMinistry = "Citizen's Services";
  const stubPropMigrationLicensePlate = 'abcdef';

  const { container } = render(
    <ProjectCard
      projectDetails={{
        name: stubPropTitle,
        description: stubPropTextBody,
        ministryName: stubPropMinistry,
        migratingLicenseplate: stubPropMigrationLicensePlate,
      }}
    />,
  );

  expect(container).toMatchSnapshot();
});
