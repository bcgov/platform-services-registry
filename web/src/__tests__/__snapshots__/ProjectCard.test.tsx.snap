// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard /> should render with Migration Licenseplate 1`] = `
<div>
  <div
    class="css-z54kij"
  >
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Project Name
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the public name of the application, please avoid acronyms;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Health Gateway
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Project Description
      </h3>
      <p
        class="css-18ja79e"
      >
        This is a brief description of your project;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        This is a test description for health gateway app
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Ministry
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the ministry that is responsible for your application;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Citizen's Services
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Cluster
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the cluster this namespace is provisioned in;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      />
    </div>
  </div>
</div>
`;

exports[`<ProjectCard /> should render without Migration Licenseplate 1`] = `
<div>
  <div
    class="css-z54kij"
  >
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Project Name
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the public name of the application, please avoid acronyms;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Health Gateway
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Project Description
      </h3>
      <p
        class="css-18ja79e"
      >
        This is a brief description of your project;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        This is a test description for health gateway app
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Ministry
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the ministry that is responsible for your application;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Citizen's Services
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Cluster
      </h3>
      <p
        class="css-18ja79e"
      >
        This is the cluster this namespace is provisioned in;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      />
    </div>
  </div>
</div>
`;
